File: InsertSort.java
public class InsertSort {


    void sort(int[] tab){
        for (int i = 1; i < tab.length; ++i) {
            int key = tab[i], j = i - 1;

            while (j >= 0 && tab[j] > key) {
                tab[j + 1] = tab[j];
                j--;
            }

            tab[j + 1] = key;
        }


    }
}

File: insertSort.py

def insertSort(inputTab):
    for lap in range(1,len(inputTab)):
        for i in range(lap,0,-1):
            if inputTab[i] < inputTab[i-1]:
                inputTab[i-1],inputTab[i] = inputTab[i], inputTab[i-1]
            else:
                break
